src/coord/src/coord.rs

```diff
@@ -5224,6 +5230,15 @@ impl Coordinator {
         let df = DataflowDesc::new("".into());
         let _: () = self.ship_dataflow(df.clone()).await;
         let _: () = self.ship_dataflows(vec![df.clone()]).await;
-        let _: DataflowDescription<mz_dataflow_types::plan::Plan> = self.finalize_dataflow(df);
+        let _: DataflowDescription<mz_dataflow_types::plan::Plan> =
+            self.finalize_dataflow(df.clone());
+
+        // finalize_dataflow is called by sequence_peek but doesn't actually
+        // ship the dataflow. We thus want finalize_dataflow to not have any side
+        // effects itself, and so it should not take a `&mut self`, but instead a
+        // `&self`. Prevent folks from changing this in the future with a compilation
+        // error should they try to make it mut again.
+        let no_mut: &Coordinator = &self;
+        no_mut.finalize_dataflow(df.clone());
```

Comment:
This is a janky test function that is meant to prevent us programmers from changing the function signatures here. The call above is ensuring that the return type is a non-Result (co-located with other things that also can't return Result), and the "call" here is to ensure this function doesn't get made mut again.

I'm 100% onboard people telling me this test function is silly and not useful, but none of this function ever runs.

Reply:
Ah sorry. I fell victim to github collapsing the surrounding stuff and didn't investigate whether the code would be called.